pkg/manager/manager_tasks.go

```diff
@@ -99,7 +99,13 @@ func (s *singleton) Scan(useFileMetadata bool) {
 		for _, path := range config.GetStashPaths() {
 			globPath := filepath.Join(path, "**/*."+constructGlob())
 			globResults, _ := doublestar.Glob(globPath)
-			results = append(results, globResults...)
+
+			for _, globCheck := range globResults {
```

Comment:
Imo checking for directories shouldn't happen here. It adds a lot of os.stat calls and thus delay for users with network mounted dirs (or even worse gdrive mounts ).
I would check for dirs/filter the results  in the task_scan.go file, when scaning the scene/gallery. 
If it is done there after the db findbypath 
-  if the path is in the DB  it would filter out all the files that are already in the db since directories shouldnt be there anyway due to checksum calculation error
- if the path isnt it would checksum it anyway so the os.stat doesn't `cost` us

Reply:
Thanks @bnkai for the excellent idea of doing it further down in the scanning process. Always felt this change added a huge amount of overhead for a very minor issue, caused by an edge case. I'll rework it.

Reply:
Reworked.